config/src/lib.rs

```diff
@@ -621,7 +622,18 @@ impl Config {
     pub fn to_string_pretty(&self) -> Result<String, toml::ser::Error> {
         // serializing to value first to prevent `ValueAfterTable` errors
         let value = toml::Value::try_from(self)?;
-        let s = toml::to_string_pretty(&value)?;
+        let mut s = toml::to_string_pretty(&value)?;
+
+        if self.optimizer_details.is_some() {
+            // this is a hack to make nested tables work because this requires the config's profile
+            s = s
+                .replace("[optimizer_details]", &format!("[{}.optimizer_details]", self.profile))
+                .replace(
+                    "[optimizer_details.yulDetails]",
+                    &format!("[{}.optimizer_details.yulDetails]", self.profile),
+                );
+        }
```

Comment:
does this make it work even if the profile is not set?

Reply:
this is only really used in `forge config`, and there's always a profile, if not set via `FOUNDRY_PROFILE` it's always [default]